quaint/src/connector/mysql/native/mod.rs

```diff
@@ -0,0 +1,380 @@
+                        break;
+                    }
+                }
+
+                Ok(())
+            })
+            .await
+        })
+        .await
+    }
+
+    async fn version(&self) -> crate::Result<Option<String>> {
+        let query = r#"SELECT @@GLOBAL.version version"#;
+        let rows = timeout::socket(self.socket_timeout, self.query_raw(query, &[])).await?;
+
+        let version_string = rows
+            .get(0)
+            .and_then(|row| row.get("version").and_then(|version| version.typed.to_string()));
+
+        Ok(version_string)
+    }
+
+    fn is_healthy(&self) -> bool {
+        self.is_healthy.load(Ordering::SeqCst)
+    }
+
+    async fn set_tx_isolation_level(&self, isolation_level: IsolationLevel) -> crate::Result<()> {
+        if matches!(isolation_level, IsolationLevel::Snapshot) {
+            return Err(Error::builder(ErrorKind::invalid_isolation_level(&isolation_level)).build());
+        }
+
+        self.raw_cmd(&format!("SET TRANSACTION ISOLATION LEVEL {isolation_level}"))
+            .await?;
+
+        Ok(())
+    }
+
+    fn requires_isolation_first(&self) -> bool {
+        true
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::MysqlUrl;
+    use crate::tests::test_api::mysql::CONN_STR;
+    use crate::{error::*, single::Quaint};
+    use url::Url;
+
+    #[test]
+    fn should_parse_socket_url() {
+        let url = MysqlUrl::new(Url::parse("mysql://root@localhost/dbname?socket=(/tmp/mysql.sock)").unwrap()).unwrap();
+        assert_eq!("dbname", url.dbname());
+        assert_eq!(&Some(String::from("/tmp/mysql.sock")), url.socket());
+    }
+
+    #[test]
+    fn should_parse_prefer_socket() {
+        let url =
+            MysqlUrl::new(Url::parse("mysql://root:root@localhost:3307/testdb?prefer_socket=false").unwrap()).unwrap();
+        assert!(!url.prefer_socket().unwrap());
+    }
+
+    #[test]
+    fn should_parse_sslaccept() {
+        let url =
+            MysqlUrl::new(Url::parse("mysql://root:root@localhost:3307/testdb?sslaccept=strict").unwrap()).unwrap();
+        assert!(url.query_params.use_ssl);
+        assert!(!url.query_params.ssl_opts.skip_domain_validation());
+        assert!(!url.query_params.ssl_opts.accept_invalid_certs());
+    }
+
+    #[test]
+    fn should_parse_ipv6_host() {
+        let url = MysqlUrl::new(Url::parse("mysql://[2001:db8:1234::ffff]:5432/testdb").unwrap()).unwrap();
+        assert_eq!("2001:db8:1234::ffff", url.host());
+    }
+
+    #[test]
+    fn should_allow_changing_of_cache_size() {
+        let url = MysqlUrl::new(Url::parse("mysql:///root:root@localhost:3307/foo?statement_cache_size=420").unwrap())
+            .unwrap();
+        assert_eq!(420, url.cache().capacity());
+    }
+
+    #[test]
+    fn should_have_default_cache_size() {
+        let url = MysqlUrl::new(Url::parse("mysql:///root:root@localhost:3307/foo").unwrap()).unwrap();
+        assert_eq!(100, url.cache().capacity());
+    }
```

Comment:
These tests should most likely be part of the url.rs crate

Reply:
It might be the case for other connectors

Reply:
Thanks, I've addressed this in https://github.com/prisma/prisma-engines/pull/4442/commits/95a4e28c89a1cda3c7bdf6f15b7ae543ae5da780.